Cython/Compiler/Parsing.py

```diff
@@ -2477,13 +2477,11 @@ def p_positional_and_keyword_args(s, end_sy_set, templates = None):
         s.next()
     return positional_args, keyword_args
 
-def p_c_base_type(s, self_flag = 0, nonempty = 0, templates = None):
-    # If self_flag is true, this is the base type for the
-    # self argument of a C method of an extension type.
+def p_c_base_type(s, nonempty = 0, templates = None):
```

Comment:
Whenever we touch such code, it's a good time to clean it up along the way.
```suggestion
def p_c_base_type(s, nonempty=False, templates=None):
```
Note that you need to change `Parsing.pxd` as well (we should have a test job that exercises the `--cython-compile-more` build option – would also be interesting to see if it makes a performance dfference).

Reply:
I'm working on adding CI jobs. Seems we've piled up some more differences in the past already.